eox_core/social_tpa_backends.py

```diff
@@ -61,3 +63,44 @@ def extra_data(self, user, uid, response, details=None, *args, **kwargs):  # pyl
         data["user_details"] = self.get_user_details(response)
 
         return data
+
+    def get_user_id(self, *args, **kwargs):
+        """
+        Insert a slug at the beginning of the user id. Retroactively add the
+        slug for social users that don't have one. The slug must be defined
+        in the 'OTHER_SETTINGS' field from the provider configuration.
+        In case a slug was not defined raise an exception due to misconfigured
+        backend.
+        This three behaviours can be controlled via the settings:
+            - 'SOCIAL_AUTH_NAMESPACED_UIDS': if false behave as originally
+            intended. If true, add a slug to the uid.
+            - 'SOCIAL_AUTH_ALLOW_SLUGLESS_UID': if false raise and exception when no slug
+            is defined. If true, return de original uid when no slug is defined.
+            - 'SOCIAL_AUTH_ALLOW_WRITE_SLUG_UID': If true update the uid of existing social
+            users if any.
+        """
+        uid = super().get_user_id(*args, **kwargs)
+        namespaced_uids = self.setting('NAMESPACED_UIDS', False)
+        if not namespaced_uids:
+            return uid
+
+        strategy = self.strategy
+        slug = strategy.setting('slug', backend=self)
+        provider = self.name
+        allow_slugless_uid = self.setting('ALLOW_SLUGLESS_UID', False)
+        allow_write_slug_uid = self.setting('ALLOW_WRITE_SLUG_UID', True)
+
+        if not slug:
+            if allow_slugless_uid:
+                return uid
+            raise AuthMissingParameter(self, 'slug')
```

Comment:
This is what happens when I dont put anything in the settings right?

Does this mean the change breaks stuff and we need to bump major? I'm not convinced we need to, but I dont know how do we mark this in the code, the repo or the Changelog.

Alternatively, we make it so not adding anything to settings makes it behave exactly as it does now.

Reply:
This will only happen if you set `'SOCIAL_AUTH_NAMESPACED_UIDS'` to `true` but don't define a slug in `OTHER_SETTINGS`. It won't break anything unless you are making a conscious effort to use this feature but don't configure it correctly.

Reply:
Then it is ok